src/youdo/core.clj

```diff
@@ -1,24 +1,31 @@
 (ns youdo.core
-  (:require [youdo.state :as state]
-            [clojure.tools.cli :as cli-tools])
+  (:require [clojure.tools.cli :as cli-tools]
+            [clojure.string :as str]
+            [youdo.state :as state])
   (:gen-class))
 
+(defonce supported-actions #{"add" "list"})
+
 (def cli-options
-  ;; An option with a required argument
   [["-t" "--task TASK" "Task description"]
    ["-p" "--path PATH" "Add path"
-    :default "default-path"]
-   ;; A boolean option defaulting to nil
-   ["-h" "--help"]])
+    :default "default-path"]])
 
 (defn parse-args [action args]
   (let [opts (cli-tools/parse-opts args cli-options)
-        task (get-in opts [:options :task])
-        path (get-in opts [:options :path])]
-    (cond-> {:action action
-             :path path}
+        task (get-in opts [:options :task])]
+
+    (assert (not (str/blank? action)) "Action not found")
+    (assert (not (and (= action "add") (nil? task))) "Task not found")
+    (assert (supported-actions action) "Incorrect action")
```

Comment:
This check alone would fail if the action was empty! You can merge the first assert with this one